renderer/reducers/autopilot.js

```diff
@@ -0,0 +1,74 @@
+import { grpcService } from 'workers'
+import { requestNodeScores } from '@zap/utils/api'
+import { infoSelectors } from './info'
+
+// ------------------------------------
+// Constants
+// ------------------------------------
+
+export const UPDATE_AUTOPILOT_NODE_SCORES = 'UPDATE_AUTOPILOT_NODE_SCORES'
+export const UPDATE_AUTOPILOT_NODE_SCORES_SUCCESS = 'UPDATE_AUTOPILOT_NODE_SCORES_SUCCESS'
+export const UPDATE_AUTOPILOT_NODE_SCORES_FAILURE = 'UPDATE_AUTOPILOT_NODE_SCORES_FAILURE'
+
+// ------------------------------------
+// Actions
+// ------------------------------------
+
+// Fetch autopilot node scores.
+export const updateAutopilotNodeScores = () => async (dispatch, getState) => {
+  const state = getState()
+
+  // Only apply node scores to local nodes where autopilot is enabled.
+  const { type, autopilot } = state.lnd.lndConfig
+  if (type !== 'local' || !autopilot) {
+    return
+  }
+
+  try {
+    dispatch({ type: UPDATE_AUTOPILOT_NODE_SCORES })
+
+    const chain = infoSelectors.chainSelector(state)
+    const network = infoSelectors.networkSelector(state)
+    const scores = await requestNodeScores(chain, network)
+    if (!scores) {
+      throw new Error('Node scores empty')
```

Comment:
I'm not sure there is any need. If this throws an error because there are no scores, the `UPDATE_AUTOPILOT_NODE_SCORES_FAILURE` action will be fired and the error detail will be included in that. So you can see the details in the logs already.

In the future, we could consider doing something other than just storing the error in the redux store such adding an entry to a global notification system (which we don't have yet).

As per your other comment below, I have added a warning message when preprocessing the scores in the case that we encounter invalid data.